src/ros2_gamepad_to_twist_message/ros2_gamepad_to_twist_message/playstation.py

```diff
@@ -69,8 +70,16 @@ def timer_callback(self):
         # initializes Twist message
         twist = Twist()
 
-        # retrieve any events from the controller
         for event in controller.event.get():
+            if event.type == controller.JOYDEVICEADDED:
+                joy = controller.joystick.Joystick(event.device_index)
+                self.joysticks[joy.get_instance_id()] = joy
+                self.get_logger().info(f"Joystick {joy.get_instance_id()} connected")
```

Comment:
don't you want to use your newly made self.joysticks.get_instanced_id instead of joy.get_instance_id?